diff --git a/Dockerfile b/Dockerfile
index 512e2d17..1e3c7498 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -8,7 +8,9 @@ RUN apt update && apt install lld clang -y
 FROM chef as planner
 COPY . .
 # Compute a lock-like file for our project
-RUN cargo chef prepare --recipe-path recipe.json
+RUN mkdir -p ~/.cargo/
+COPY ./.cargo-config ~/.cargo/config
+RUN cargo chef prepare --recipe-path recipe.json; git config --global url."https://kkgithub.com/".insteadOf https://github.com/
 
 FROM chef as builder
 
@@ -22,6 +24,8 @@ ARG PROFILE="release"
 COPY --from=planner /app/recipe.json recipe.json
 # Build our project dependencies
 ENV CARGO_BUILD_JOBS=4
+RUN mkdir -p ~/.cargo/; git config --global url."https://kkgithub.com/".insteadOf https://github.com/
+COPY ./.cargo-config ~/.cargo/config
 RUN cargo chef cook --release --recipe-path recipe.json
 
 COPY . .
@@ -29,7 +33,9 @@ ENV SQLX_OFFLINE true
 
 # Build the project
 RUN echo "Building with profile: ${PROFILE}, features: ${FEATURES}, "
-RUN cargo build --profile=${PROFILE} --features "${FEATURES}" --bin appflowy_cloud
+RUN mkdir -p ~/.cargo/
+COPY ./.cargo-config ~/.cargo/config
+RUN git config --global url."https://kkgithub.com/".insteadOf https://github.com/; RUSTFLAGS="-C link-arg=-fuse-ld=lld" cargo build --jobs 1 --profile=${PROFILE} --features "${FEATURES}" --bin appflowy_cloud
 
 FROM debian:bookworm-slim AS runtime
 WORKDIR /app
diff --git a/docker/gotrue/Dockerfile b/docker/gotrue/Dockerfile
index 100b6b8c..407dbd7f 100644
--- a/docker/gotrue/Dockerfile
+++ b/docker/gotrue/Dockerfile
@@ -1,11 +1,12 @@
 # syntax=docker/dockerfile:1
 FROM golang as base
 WORKDIR /go/src/supabase
-RUN git clone https://github.com/supabase/auth.git --depth 1 --branch v2.159.1
+RUN git config --global url."https://kkgithub.com/".insteadOf https://github.com/; \
+ git clone https://github.com/supabase/auth.git --depth 1 --branch v2.159.1
 WORKDIR /go/src/supabase/auth
 COPY patch/mfa_enabled.patch .
 RUN git apply mfa_enabled.patch
-RUN CGO_ENABLED=0 go build -o /auth .
+RUN git config --global url."https://kkgithub.com/".insteadOf https://github.com/; CGO_ENABLED=0 GOPROXY=https://goproxy.io,direct go build -o /auth .
 RUN rm /go/src/supabase/auth/migrations/20240612123726_enable_rls_update_grants.up.sql
 
 FROM alpine:3.20
